src/dotnet/Program.cs

```diff
@@ -178,6 +182,9 @@ private static void ConfigureDotNetForFirstTimeUse()
                             environmentProvider);
 
                         dotnetConfigurer.Configure();
+
+                        if (!File.Exists(Telemetry.TelemetrySentinel))
+                            File.Create(Telemetry.TelemetrySentinel);
```

Comment:
Printing the telemetry eula is being clubbed with nuget cache priming, hence this eula will get printed again and again if nuget cache prime fails, since nuget cache sentinel is set only after restore succeeds. Do we want to hve this behavior? or separate the eula from nuget cache priming?